github/event_types_test.go

```diff
@@ -9560,1789 +9563,68 @@ func TestReleaseEvent_Marshal(t *testing.T) {
+		NodeID:   Ptr("n"),
+		Name:     Ptr("r"),
+		FullName: Ptr("o/r"),
+	}
 
-	testJSONMarshal(t, u, want)
+	sender := &User{
+		ID:     Ptr(int64(1)),
+		NodeID: Ptr("n"),
+		Login:  Ptr("l"),
+	}
+
+	tests := []struct {
+		name  string
+		json  string
+		event *RepositoryRulesetEvent
+	}{
+		{"empty", `{}`, &RepositoryRulesetEvent{}},
+		{"created", fmt.Sprintf(`{"action":"created","repository_ruleset":{"id":1,"name":"r","target":"branch","source_type":"Repository","source":"o/r","enforcement":"active","conditions":{"ref_name":{"exclude":[],"include":["~ALL"]}},"rules":[{"type":"deletion"},{"type":"creation"},{"type":"update"},{"type":"required_linear_history"},{"type":"pull_request","parameters":{"required_approving_review_count":2,"dismiss_stale_reviews_on_push":false,"require_code_owner_review":false,"require_last_push_approval":false,"required_review_thread_resolution":false,"automatic_copilot_code_review_enabled":false,"allowed_merge_methods":["squash","rebase","merge"]}},{"type":"code_scanning","parameters":{"code_scanning_tools":[{"tool":"CodeQL","security_alerts_threshold":"high_or_higher","alerts_threshold":"errors"}]}}],"node_id":"n","created_at":%[1]s,"updated_at":%[1]s,"_links":{"self":{"href":"a"},"html":{"href":"a"}}},"repository":{"id":1,"node_id":"n","name":"r","full_name":"o/r"},"organization":{"id":1,"node_id":"n","name":"o"},"enterprise":{"id":1,"node_id":"n","slug":"e","name":"e"},"installation":{"id":1,"node_id":"n","app_id":1,"app_slug":"a"},"sender":{"id":1,"node_id":"n","login":"l"}}`, referenceTimeStr), &RepositoryRulesetEvent{Action: Ptr("created"), RepositoryRuleset: &RepositoryRuleset{ID: Ptr(int64(1)), Name: "r", Target: Ptr(RulesetTargetBranch), SourceType: Ptr(RulesetSourceTypeRepository), Source: "o/r", Enforcement: RulesetEnforcementActive, Conditions: &RepositoryRulesetConditions{RefName: &RepositoryRulesetRefConditionParameters{Include: []string{"~ALL"}, Exclude: []string{}}}, Rules: &RepositoryRulesetRules{Creation: &EmptyRuleParameters{}, Update: &UpdateRuleParameters{}, Deletion: &EmptyRuleParameters{}, RequiredLinearHistory: &EmptyRuleParameters{}, PullRequest: &PullRequestRuleParameters{AllowedMergeMethods: []MergeMethod{MergeMethodSquash, MergeMethodRebase, MergeMethodMerge}, DismissStaleReviewsOnPush: false, RequireCodeOwnerReview: false, RequireLastPushApproval: false, RequiredApprovingReviewCount: 2, RequiredReviewThreadResolution: false}, CodeScanning: &CodeScanningRuleParameters{CodeScanningTools: []*RuleCodeScanningTool{{AlertsThreshold: CodeScanningAlertsThresholdErrors, SecurityAlertsThreshold: CodeScanningSecurityAlertsThresholdHighOrHigher, Tool: "CodeQL"}}}}, NodeID: Ptr("n"), CreatedAt: &Timestamp{referenceTime}, UpdatedAt: &Timestamp{referenceTime}, Links: &RepositoryRulesetLinks{Self: &RepositoryRulesetLink{HRef: Ptr("a")}, HTML: &RepositoryRulesetLink{HRef: Ptr("a")}}}, Repository: repository, Organization: organization, Enterprise: enterprise, Installation: installation, Sender: sender}},
```

Comment:
This line contains 2668 characters, which may make it difficult to edit in the future as fields change. Could you please format the JSON as it is done in the other tests in this file?

Reply:
@alexandear I'm going to have to disagree with you on this. As the tests should be tightly coupled with the JSON string it makes the most sense to make the JSON format as clear and unambiguous as possible. Therefore the compact representation is the obvious choice. If you want to edit the JSON it's going to be much easier to do so by copying the string and working on it as actual JSON (`jq -c '.'` will then format the JSON to be pasted back).

Please note that this test doesn't use `testJSONMarshal` like the other tests; that helper function can (and does) mask incorrect JSON by depending on the very marshaling code that's under test to format the JSON string.